genum/gen/generate.go

```diff
@@ -263,3 +307,18 @@ func isDeprecated(fAST *ast.File, name string) bool {
 	}
 	return false
 }
+
+func (g *Generate) ConvertibleFromString(inputType types.Type) bool {
+	return types.ConvertibleTo(stringType, inputType)
+}
+
+func (g *Generate) ConvertibleFromInt(inputType types.Type) bool {
+	// you can get an int from an untyped int and an untyped string
+	// but we don't want them to return true because
+	if basicType, ok := inputType.(*types.Basic); ok {
+		if basicType.Kind() == types.UntypedInt || basicType.Kind() == types.UntypedString {
+			return false
+		}
```

Comment:
An untyped int is just `int`.. so something isn't quite right here. You can add a trait to tests to validate this. 

I also suspect that this particular mechanism just isn't expressive enough to handle all the variations of a single type. You might want something like: 

```go
func extractUnderlying(p types.Type) (string, bool) {
	v, ok := p.Underlying().(*types.Basic)
	if !ok {
		return "", false
	}
	switch v.Kind() {
	case
		types.UntypedInt,
		types.Int,
		types.Int8,
		types.Int16,
		types.Int32,
		types.Int64:
		return "int64", true
	case types.Uint,
		types.Uint8,
		types.Uint16,
		types.Uint32,
		types.Uint64:
		return "uint64", true
	case types.Float32:
		return "float32", true
	case types.UntypedFloat, types.Float64:
		return "float64", true
	case types.UntypedString, types.String:
		return "string", true
	}
	return "invalid", true
}

```

For this use case specifically.

Reply:
I added this functionality but its definitely not possible to have `UntypedString` and `UntypedInt`. The generated code  doesnt compile/make sense when they are included